lib/clashom.gi

```diff
@@ -341,7 +341,7 @@ local clT,        # classes T
           cen:=centralizers[t];
 
           if not IsBound(etas[t]) then
-            if Number(etas,i->IsBound(i))>500 then
+            if Number(etas)>500 then
```

Comment:
Just also changed this `Number` call and the one later in the file; I verified that `Number(list)` indeed counts the number of *bound* entries in `list`, so the change is correct.

That leaves one use of `x->IsBound(x)` in GAP (at least that I could find):
```
$ git grep -n -P '([a-zA-Z]+)\s*->\s*IsBound\(\1\)'
tst/testinstall/syntaxtree.tst:2954:gap> testit(x -> IsBound(x));
```
But there it is used to test the syntaxtree feature, so we should retain it.